src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -4,204 +4,131 @@
 
 using System.Collections;
 using System.Diagnostics;
-using System.Runtime.CompilerServices;
 using System.Text.Json.Serialization;
 
 namespace System.Text.Json
 {
+    [DebuggerDisplay("ClassType.{JsonClassInfo.ClassType}, {JsonClassInfo.Type.Name}")]
     internal struct WriteStackFrame
     {
-        // The object (POCO or IEnumerable) that is being populated.
-        public object? CurrentValue;
-        public JsonClassInfo? JsonClassInfo;
-
-        // Support Dictionary keys.
-        public string? KeyName;
-
-        // The current IEnumerable or IDictionary.
+        /// <summary>
+        /// The enumerator for resumable collections.
+        /// </summary>
         public IEnumerator? CollectionEnumerator;
-        // Note all bools are kept together for packing:
-        public bool PopStackOnEndCollection;
-
-        // The current object.
-        public bool PopStackOnEndObject;
-        public bool StartObjectWritten;
-        public bool MoveToNextProperty;
-
-        public bool WriteWrappingBraceOnEndPreservedArray;
 
-        // The current property.
-        public int PropertyEnumeratorIndex;
-        public ExtensionDataWriteStatus ExtensionDataStatus;
-        public JsonPropertyInfo? JsonPropertyInfo;
-
-        // Pre-encoded metadata properties.
-        private static readonly JsonEncodedText s_metadataId = JsonEncodedText.Encode("$id", encoder: null);
-        private static readonly JsonEncodedText s_metadataRef = JsonEncodedText.Encode("$ref", encoder: null);
-        private static readonly JsonEncodedText s_metadataValues = JsonEncodedText.Encode("$values", encoder: null);
-
-        public void Initialize(Type type, JsonSerializerOptions options)
+        /// <summary>
+        /// The original JsonPropertyInfo that is not changed. It contains all properties.
+        /// </summary>
+        /// <remarks>
+        /// For objects, it is either the policy property for the class or the current property.
+        /// For collections, it is either the policy property for the class or the policy property for the current element.
+        /// </remarks>
+        public JsonPropertyInfo DeclaredJsonPropertyInfo;
+
+        /// <summary>
+        /// Used when processing dictionaries.
+        /// </summary>
+        public bool IgnoreDictionaryKeyPolicy;
+
+        /// <summary>
+        /// The class (POCO or IEnumerable) that is being populated.
+        /// </summary>
+        public JsonClassInfo JsonClassInfo;
+
+        /// <summary>
+        /// The key name for a dictionary value.
+        /// </summary>
+        public string KeyName;
```

Comment:
This is no longer used.

Reply:
Thanks. Missed it when converted to use JsonPropertyNameAsString.